### DISCUSSION TITLE
Recommendations for lit-localizing a monorepo with multiple packages?

### DISCUSSION THREAD

**Discussion by kensternberg-authentik** [lit/lit#4883] [Q&A]
We use lit-localize a lot at [authentik](https://goauthentik.io), and our product is now at the level of complexity where it makes some sense to start breaking it up into a monorepo.  We have three major webapps, all written with Lit, and we'd like to make packages for each of them, as well as a package for our shared web components.

We're stymied by lit-localize, though.  We're not sure how to manage our complete collection, and that's more than a little frustrating.  Our translation service is pushing back (well, threatening to charge more) if we submit multiple translation files per language (as in, one-per-app), and since there's language in both an application and the web-components, we'd like not to have to manage multiple contexts per app, especially not ones namespaced for, say "elements" and "app-admin".

Our (current) solution is to move the context handler and definition files into the `./packages/locales` package, but run the extract and build from the root, pushing the results into the locales package, as an early build stage.  But that's kinda gross; it means our Locales package is little more than a holder of build scripts.

Does anyone have a solid recommendation for dealing with lit-localize in a monorepo set-up like this one?

**Comment by augustjk** [comment]
this is indeed a sore point for lit localize. there have been some thoughts around this here https://github.com/lit/lit/issues/1661 but unfortunately we never got around to it. at this point, i can't say if/when that work could happen without significant community contribution.
